src/adapter/bun/websocket.ts

```diff
@@ -0,0 +1,115 @@
+// @denoify-ignore
+import {
+  createWSMessageEvent,
+  type UpgradeWebSocket,
+  type WSContext,
+  type WSEvents,
+  type WSMessageReceive,
+} from '../../helper/websocket'
+
+interface BunServerWebSocket<T> {
+  send(data: string | ArrayBufferLike, compress?: boolean): void
+  close(code?: number, reason?: string): void
+  data: T
+  readyState: 0 | 1 | 2 | 3
+}
+interface BunServer {
+  upgrade<T>(
+    req: Request,
+    options?: {
+      data: T
+    }
+  ): boolean
+}
+interface BunWebSocketHandler<T> {
+  open(ws: BunServerWebSocket<T>): void
+  close(ws: BunServerWebSocket<T>, code?: number, reason?: string): void
+  message(ws: BunServerWebSocket<T>, message: string | Buffer): void
+}
+interface CreateBunWebSocket {
+  (): {
+    upgradeWebSocket: UpgradeWebSocket
+    websocket: BunWebSocketHandler<BunWebSocketData>
+  }
+}
+export interface BunWebSocketData {
+  connId: number
+  url: URL
+  protocol: string
+}
+
+const createWSContext = (ws: BunServerWebSocket<BunWebSocketData>): WSContext => {
+  return {
+    send: (source, options) => {
+      const sendingData =
+        typeof source === 'string' ? source : source instanceof Uint8Array ? source.buffer : source
+      ws.send(sendingData, options?.compress)
+    },
+    raw: ws,
+    binaryType: 'arraybuffer',
+    readyState: ws.readyState,
+    url: ws.data.url,
+    protocol: ws.data.protocol,
+    close(code, reason) {
+      ws.close(code, reason)
+    },
+  }
+}
+
+export const createBunWebSocket: CreateBunWebSocket = () => {
```

Comment:
I think it does not need `Bun` in the name. So it can be `createWebSocket`.

The user will know if it is for Bun because it is imported from `'hono/ws/bun'`.

Reply:
I'm reconsidering. `BunWebSocket` might be a good name for this since it contains Bun specific objects.